File: atividade-4/025_menores/025_menores.js
// Concatenar dois vetores

const number = 3;
const list = [1, 3, 6, 7, 9, 3];

const minimum = (xs) => Math.min(...xs);
const maximum = (xs) => Math.max(...xs);
const listGen = (min, max) =>
  [...Array(max + 1).keys()].filter((x) => x > min - 1);

const menores = (n, xs) => {
  if (!xs.length) return [];

  if (xs.length === n) return xs;
  if (maximum(xs) === minimum(xs)) return listGen(xs.length - n, xs.length);

  return menores(
    n,
    xs.filter((x) => x !== maximum(xs))
  );
};

console.log(menores(number, list));

File: atividade-4/025_menores/025_menores.hs
module Main where

-- Concatenar dois vetores

menores :: Int -> [Int] -> [Int]
menores 0 xs = []
menores n xs
      | length xs == n = xs
      | maximum xs == minimum xs = [(length xs - n)..(length xs)]
      | otherwise = menores n [x | x <- xs, x /= maximum xs]

main = do
  a <- readLn :: IO Int
  b <- readLn :: IO [Int]
  print $ menores a b